src/uniqlist.c

```diff
@@ -356,16 +367,27 @@ SEXP uniqueNlogical(SEXP x, SEXP narmArg) {
   R_xlen_t i=0;
   const int *ix = LOGICAL(x);
   while (++i<n && ix[i]==first);
-  if (i==n)
-    return ScalarInteger(first==NA_INTEGER && narm ? 0 : 1); // all one value
+  if (i==n) { // all values the same
+    if (first==NA_INTEGER && narm) {
+      return ScalarInteger(0);
+    } else {
+      return ScalarInteger(1);
+    }
+  }
   Rboolean second = ix[i];
   // we've found 2 different values (first and second). Which one didn't we find? Then just look for that.
   // NA_LOGICAL == INT_MIN checked in init.c
-  const int third = (first+second == 1) ? NA_LOGICAL : ( first+second == INT_MIN ? TRUE : FALSE );
+  int third;
```

Comment:
this is not a `const` anymore, but to address compiler warnings I would need to `int tmp; if (...) tmp = ... else tmp = ...; const int third = tmp;` so I think it is OK